src/Bicep.Core/Registry/ModuleRegistry.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Modules;
+
+namespace Bicep.Core.Registry
+{
+    public abstract class ModuleRegistry<T> : IModuleRegistry where T : ModuleReference
```

Comment:
Created this to avoid calling `ConvertReference` in subclasses.

Reply:
Yeah, thanks for doing it!